### ISSUE TITLE
Uncoverd files don't show up in report or decrease overall coverage

### ISSUE THREAD

**Issue by beyerleinf** [istanbuljs/nyc#1213]
## Link to bug demonstration repository
[nyc-typescript-coverage-issue](https://github.com/beyerleinf/nyc-typescript-coverage-issue)

## Expected Behavior
Files that don't have any coverage at all (i.e. no test file) should show up in the report and add to total blocks etc. - thus decreasing overall coverage for a project.

## Observed Behavior
Files without tests don't get included in coverage reports even though `all` is set to true in .nycrc.json

### Troubleshooting steps
- [x] still occurring when I put `cache: false` in my nyc config

## Environment Information

```
  System:
    OS: Linux 4.9 LMDE 3 (cindy) 3 (cindy)
    CPU: (6) x64 AMD Ryzen 7 2700X Eight-Core Processor
    Memory: 687.27 MB / 15.66 GB
  Binaries:
    Node: 12.3.1 - ~/.nvm/versions/node/v12.3.1/bin/node
    npm: 6.11.3 - ~/.nvm/versions/node/v12.3.1/bin/npm
  npmPackages:
    nyc: ^14.1.1 => 14.1.1 
    source-map-support: ^0.5.13 => 0.5.13 
    ts-node: ^8.4.1 => 8.4.1 
    typescript: ^3.6.4 => 3.6.4 

```

**Comment by coreyfarrell** [comment]
For nyc 14 you need to add `"extension": [".ts"]` to your `.nycrc.json`.

**Comment by beyerleinf** [comment]
That's why my `.nycrc.json` uses [@istanbuljs/nyc-config-typescript](https://www.npmjs.com/package/@istanbuljs/nyc-config-typescript). It's already set there (see [Line 4 of index.json at nyc-config-typescript](https://github.com/istanbuljs/istanbuljs/blob/70f48b7e6b60d17b8e2b5a03388a004deb6b8785/packages/nyc-config-typescript/index.json#L4))

**Comment by coreyfarrell** [comment]
Hmmm good point sorry I missed that.  Could you try adding `parser-plugins` to your `.nycrc.json`?  Specifically please set: `"parser-plugins": ["typescript"]`.  If that doesn't work please try:
```js
  "parser-plugins": [
    "typescript",
    "asyncGenerators",
    "bigInt",
    "classProperties",
    "classPrivateProperties",
    "dynamicImport",
    "importMeta",
    "objectRestSpread",
    "optionalCatchBinding"
  ]
```

I suspect the issue is that your `src` directory contains typescript syntax which is not transpiled during the `--all` stage, so istanbul-lib-instrument encounters syntax which it cannot parse.

**Comment by beyerleinf** [comment]
`"parser-plugins": ["typescript"]` did the trick. But I think adding the others won't hurt. I just noticed that I know encounter the race condition from #1113 but I suspect there is no way around that, is there?

**Comment by coreyfarrell** [comment]
nyc 15.0.0-beta.0 should fix the race condition from #1113.  There is no solution to that issue for nyc 14.

**Comment by beyerleinf** [comment]
Okay, thank you. Good to know! :)
